### PR TITLE
Avast ye string handlers

### PR DESCRIPTION
**Pull request by jcorrado**
Hi @devth,

Friday afternoon puttering!  This is partly for the fun of it, and partly to kick around some ideas for #740, and general codebase exploration.

```
jereme [4:32 PM]
!complete Why is Clojure | head 5 | xargs pirate

yetibot APP [4:32 PM]
why be cllojurre jolllly good
why be clojure popular
why be clojurre dynaamicalllly typed, yo ho ho
why be clojurre so slow, arrrrrrr
why be clojurre not popullaar, llock 'im in irrons
```

### CODE REVIEW

**Comment by devth** [comment]
Lol! Amazing. I don't see a diff on here yet - just some master branch merges.

**Comment by jcorrado** [comment]
Hrm, I botched this PR somehow.  Looking into it now.

**Comment by jcorrado** [comment]
Those pullll requests duty better when ye make some commits on tha brraanch aforre pushing, brring 'er alongside.

Fixed :-D

File: src/yetibot/commands/pirate.clj [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+;;
+(defn probability
+  "Return probability, by hour, for configured TZ."
+  []
+  (let [hour (-> (t/to-time-zone (t/now) (t/time-zone-for-id local-tz))
+                 t/hour)]
+    (nth (concat (repeat 8 0.95)
+                 (repeat 8 0.25)
+                 (repeat 8 0.75))
+         hour)))
```
Comment (devth) [inline src/yetibot/commands/pirate.clj]: Does this mean probability is increased/decreased for specific hours? Or just using time as a seed for randomness?

File: src/yetibot/commands/pirate.clj [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  [s prob]
+  (if (< (rand) prob)
+    (let [flavor (get flavor (-> (count flavor) rand int))]
+      (str/replace-first s
+                         #"[\.!?]*$"
+                         #(format ", %s%s" flavor %)))
+    s))
+
+(defn slurrr
+  "I'm not drunk, you're drunk."
```
Comment (devth) [inline src/yetibot/commands/pirate.clj]: 💯 docstring

File: src/yetibot/commands/pirate.clj [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  [f]
+  (fn [s]
+    (if (Character/isUpperCase (first s))
+      (str/replace-first (f s) #"." str/upper-case)
+      (f s))))
+
+(defn sub-word
+  [s]
+  (let [s-lc (str/lower-case s)]
+    (if (contains? dict s-lc)
```
Comment (devth) [inline src/yetibot/commands/pirate.clj]: Can simplify to `(get dict s-lc s)`.

File: src/yetibot/commands/pirate.clj [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    (nth (concat (repeat 8 0.95)
+                 (repeat 8 0.25)
+                 (repeat 8 0.75))
+         hour)))
+
+(defn suffix-flavor
+  "Possibly suffix random pirate flavor, for given probability."
+  [s prob]
+  (if (< (rand) prob)
+    (let [flavor (get flavor (-> (count flavor) rand int))]
```
Comment (devth) [inline src/yetibot/commands/pirate.clj]: Check out [`rand-nth`](https://clojuredocs.org/clojure.core/rand-nth)

File: src/yetibot/commands/pirate.clj (hunk 1 above)
Comment (jcorrado) [reply src/yetibot/commands/pirate.clj]: Yep - the probability increases throughout the day, and into the night.  We use the float corresponding to a given hour to determine the likelihood of slurring (`slurrr` fn) and suffixing little quips on converted text (`suffix-flavor` fn).  The net effect is the translations to "pirate" get more surely and incoherent as the day progresses.  It's crude, but effective... such is the pirate's life!

File: resources/pirate/flavor.edn [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ "yo ho ho"
+ "yarrr"
+ "eh"
+ "arrrghhh"
+ "arrr"
+ "ahoy matey"
+ "prepare to be boarded"
+ "hoist the mizzen"
+ "blow me down"
+ "swab the poop deck"
```
[the remark this answers is not in the document]
Comment (jcorrado) [reply resources/pirate/flavor.edn]: Let me add appropriate attribution/exoneration for this file :-D

**Comment by jcorrado** [comment]
Tests added.  This is my first time use Midje, which feels cleaner than clojure.test.  I'll update this PR shortly with the refactors you suggested above (thanks!).

**Comment by jcorrado** [comment]
I believe this is gtg, from my end.  Moving TZ and EDN bits would be a separate PR, with associated changes made elsewhere in the code base.

**Comment by devth** [comment]
@jcorrado looks good except there's a test failure!

**Comment by jcorrado** [comment]
Investigating test failure now.  Unclear why that passed on my system.  I wonder what I might muddied up to skew things.

**Comment by jcorrado** [comment]
I see the problem.  Ginning up a fix now...

**Comment by jcorrado** [comment]
GTG from my end.

**Pull request merged**